core/commands/commit_compare.py

```diff
@@ -236,11 +159,39 @@ def on_branch_selection(self, index):
         })
 
 
-class GsCompareCurrentFileAgainstCommand(WindowCommand, GitCommand):
-    def run(self, target_commit=None, file_path=None):
-        if not file_path:
-            file_path = self.file_path
-        self.window.run_command("gs_compare_against", {
-            "target_commit": target_commit,
-            "file_path": file_path
-        })
+class GsCompareAgainstCommand(WindowCommand, GitCommand):
+    def run(self, target_commit=None, file_path=None, current_file=False):
+        self._file_path = self.file_path if current_file else file_path
+        self._target_commit = target_commit
+        sublime.set_timeout_async(self.run_async)
+
+    def run_async(self):
+        options_array = [
+            "Branch",
+            "Reference"
+        ]
+
+        self.window.show_quick_panel(
+            options_array,
+            self.on_option_selection,
+            flags=sublime.MONOSPACE_FONT,
+            selected_index=self.quick_panel_compare_against_idx
+        )
+
+    def on_option_selection(self, index):
+        if index < 0:
```

Comment:
Very small nitpick: is there any condition under which this would be `-2` or some other negative value?  I believe we're using `if index == -1` in most places, since it has an explicit meaning in the context of a quick panel selection.  Would prefer that here if there isn't an opposing reason.